scio-core/src/main/scala/com/spotify/scio/values/BloomFilter.scala

```diff
@@ -0,0 +1,127 @@
+package com.spotify.scio.values
+
+import java.io.{ByteArrayOutputStream, InputStream, OutputStream}
+
+import com.google.common.hash.{Funnel, BloomFilter => gBloomFilter}
+import com.spotify.scio.coders.Coder
+import org.apache.beam.sdk.coders.AtomicCoder
+
+/**
+ * An immutable wrapper over Guava's Bloom Filter.
+ *
+ * Serializable as a field in Avro Schema.
+ *
+ * @see companion object [[BloomFilter]]
+ */
+case class BloomFilter[T] private (private val internal: gBloomFilter[T]
+) extends ApproxFilter[T] {
+
+  def expectedFpp: Double = internal.expectedFpp()
+
+  def mayBeContains(t: T): Boolean = internal.mightContain(t)
+
+  override def serialize: Array[Byte] = {
+    val ba = new ByteArrayOutputStream()
+    internal.writeTo(ba)
+    ba.toByteArray
+  }
+
+}
+
+object BloomFilter {
+
+  /**
+   * Coder for [[BloomFilter]]
+   */
+  implicit def coder[T](implicit f: Funnel[T]): Coder[BloomFilter[T]] =
```

Comment:
I wonder if we can abstract this so that we only need one `Coder[ApproxFilter[_]]` instance instead of specialized ones for each impl? I'm thinking you can serialize the FQDN of impl class as a string prefix, and have a constructor that takes `InputStream`?

That way:
- `MyBF.writeTo` writes: "com.myorg.MyBF", len(byte array), byte array
- `Coder[ApproxFilter[_]]` reads FQDN, then calls `new MyBF(rest of input stream)`